@xmltree/save.m

```diff
@@ -59,13 +59,11 @@
 addstring(sprintf(prolog));
 write_subtree(tree, root(tree), order);
 
-if ~isempty(fid)
-    if nargout == 1
-        addstring([]);
-        addstring(sprintf(prolog));
-        write_subtree(tree, root(tree), order);
-        varargout{1} = addstring;
-    end
+if ~isempty(fid) && nargout == 1
```

Comment:
This edit changes the implied conditions of the "else" subtly, and probably does the wrong thing - if `fid` is not empty, but `nargout` is not 1 (for instance calling the *documented* file output behavior with no output variable), it *will* call `addstring` with no arguments anyway and set varargout.

Reply:
Oops. I went too fast - thanks for noticing this!